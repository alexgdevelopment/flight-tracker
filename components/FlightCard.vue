<template>
  <v-card elevation="5" class="mt-6">
    <v-list-item
      ><v-list-item-content
        ><v-list-item-title>{{ flight.number }}</v-list-item-title
        ><v-list-item-subtitle
          >Flight code</v-list-item-subtitle
        ></v-list-item-content
      ></v-list-item
    >
    <v-list-item v-if="hasDepartureData"
      ><v-list-item-content
        ><v-list-item-title>{{
          flight.departure.airport.name
        }}</v-list-item-title
        ><v-list-item-subtitle>From</v-list-item-subtitle></v-list-item-content
      ></v-list-item
    >
    <v-list-item v-if="hasArrivalData"
      ><v-list-item-content
        ><v-list-item-title>{{ flight.arrival.airport.name }}</v-list-item-title
        ><v-list-item-subtitle>To</v-list-item-subtitle></v-list-item-content
      ></v-list-item
    >
    <v-list-item v-if="hasDepartureData"
      ><v-list-item-content
        ><v-list-item-title>{{ departureLocal }}</v-list-item-title
        ><v-list-item-subtitle
          >Scheduled Departure (Local Time)</v-list-item-subtitle
        ></v-list-item-content
      ></v-list-item
    >
    <v-list-item v-if="hasArrivalData"
      ><v-list-item-content
        ><v-list-item-title>{{ arrivalLocal }}</v-list-item-title
        ><v-list-item-subtitle
          >Scheduled Arrival (Local Time)</v-list-item-subtitle
        ></v-list-item-content
      ></v-list-item
    >
    <v-list-item v-if="flight.status === 'Arrived' && !!flight.arrival.terminal"
      ><v-list-item-content
        ><v-list-item-title>{{ flight.arrival.terminal }}</v-list-item-title
        ><v-list-item-subtitle
          >Terminal</v-list-item-subtitle
        ></v-list-item-content
      ></v-list-item
    >

    <v-list-item
      ><v-list-item-content
        ><v-list-item-title>{{ flight.status }}</v-list-item-title
        ><v-list-item-subtitle
          >Status</v-list-item-subtitle
        ></v-list-item-content
      ></v-list-item
    >
  </v-card>
</template>

<script>
  export default {
    name: 'FlightCard',
    props: {
      flight: {
        type: Object,
        required: true,
      },
    },
    computed: {
      hasArrivalData() {
        return !!this.flight.arrival
      },
      hasDepartureData() {
        return !!this.flight.departure
      },
      departureLocal() {
        return this.getLocalTime(this.flight.departure.scheduledTimeLocal)
      },
      arrivalLocal() {
        return this.getLocalTime(this.flight.arrival.scheduledTimeLocal)
      },
    },
    methods: {
      getLocalTime(localTime) {
        const date = new Date(localTime)
        return date.toTimeString()
      },
    },
  }
</script>