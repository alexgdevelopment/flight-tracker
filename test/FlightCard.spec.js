import { renderWithVuetify } from './helpers'
import { mockFlight } from './mockObjects'
import FlightCard from '@/components/FlightCard.vue'

describe('Flight Card', () => {
  let wrapper = null
  describe('if all data is passed', () => {
    beforeEach(() => {
      wrapper = renderWithVuetify(FlightCard, {
        propsData: { flight: mockFlight },
      })
    })

    it('should match the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })

    describe('should display', () => {
      test('flight number', () => {
        expect(wrapper.findByText(mockFlight.number))
      })
      test('departure airport name', () => {
        expect(wrapper.findByText(mockFlight.departure.airport.name))
      })
      test('departure scheduled local time', () => {
        expect(
          wrapper.findByText(
            new Date(mockFlight.departure.scheduledTimeLocal).toTimeString()
          )
        )
      })
      test('arrival airport name', () => {
        expect(wrapper.findByText(mockFlight.arrival.airport.name))
      })
      test('arrival scheduled local time', () => {
        expect(
          wrapper.findByText(
            new Date(mockFlight.arrival.scheduledTimeLocal).toTimeString()
          )
        )
      })
      test('flight status', () => {
        expect(wrapper.findByText(mockFlight.status))
      })
    })
  })
})
