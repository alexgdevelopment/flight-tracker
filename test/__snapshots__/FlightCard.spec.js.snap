// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight Card if all data is passed should match the snapshot 1`] = `
"<div class=\\"mt-6 v-card v-sheet theme--light elevation-5\\">
  <div tabindex=\\"-1\\" class=\\"v-list-item theme--light\\">
    <div class=\\"v-list-item__content\\">
      <div class=\\"v-list-item__title\\">AA-001</div>
      <div class=\\"v-list-item__subtitle\\">Flight code</div>
    </div>
  </div>
  <div tabindex=\\"-1\\" class=\\"v-list-item theme--light\\">
    <div class=\\"v-list-item__content\\">
      <div class=\\"v-list-item__title\\">Testing Departure</div>
      <div class=\\"v-list-item__subtitle\\">From</div>
    </div>
  </div>
  <div tabindex=\\"-1\\" class=\\"v-list-item theme--light\\">
    <div class=\\"v-list-item__content\\">
      <div class=\\"v-list-item__title\\">Testing Arrival</div>
      <div class=\\"v-list-item__subtitle\\">To</div>
    </div>
  </div>
  <div tabindex=\\"-1\\" class=\\"v-list-item theme--light\\">
    <div class=\\"v-list-item__content\\">
      <div class=\\"v-list-item__title\\">20:00:00 GMT+0300 (Източноевропейско лятно часово време)</div>
      <div class=\\"v-list-item__subtitle\\">Scheduled Departure (Local Time)</div>
    </div>
  </div>
  <div tabindex=\\"-1\\" class=\\"v-list-item theme--light\\">
    <div class=\\"v-list-item__content\\">
      <div class=\\"v-list-item__title\\">22:00:00 GMT+0300 (Източноевропейско лятно часово време)</div>
      <div class=\\"v-list-item__subtitle\\">Scheduled Arrival (Local Time)</div>
    </div>
  </div>
  <div tabindex=\\"-1\\" class=\\"v-list-item theme--light\\">
    <div class=\\"v-list-item__content\\">
      <div class=\\"v-list-item__title\\">1</div>
      <div class=\\"v-list-item__subtitle\\">Terminal</div>
    </div>
  </div>
  <div tabindex=\\"-1\\" class=\\"v-list-item theme--light\\">
    <div class=\\"v-list-item__content\\">
      <div class=\\"v-list-item__title\\">Arrived</div>
      <div class=\\"v-list-item__subtitle\\">Status</div>
    </div>
  </div>
</div>"
`;
